File: FWC(Intenship)/Avr_gcc_2018_18/IN2018_18.py
number = float(input("Enter a number: "))
a = int(number)
b = number - a  
binary_whole_part = ""
while a > 0:
    c = a % 2
    a = a // 2
    binary_whole_part = str(c) + binary_whole_part
binary_fractional_part = ""
while b > 0:
    if len(binary_fractional_part) > 15:
        break
    b *= 2
    bit = int(b)
    binary_fractional_part += str(bit)
    b -= bit
print("Binary representation:", binary_whole_part + "." + binary_fractional_part)

File: FWC(Intenship)/Avr_gcc_2018_18/IN2018_18.c
// Online C compiler to run C program online#include <stdio.h>
#include<stdio.h>
void IntegerToBinary(int num,char *binary){
        for(int i = 7; i>= 0; i--){
                binary[i] = (num % 2) + '0';
                num /= 2;
        }
        binary[8] = '\0';
       }
// MAIN PROGRAM
int main(void)
{
  float number = 27.625;
  int a = (int)number;
  float b = number - a;
  char binary_whole_part[9]="";
  char binary_fractional_part[9]="";
  IntegerToBinary(a,binary_whole_part);
  for(int i = 0; i < 8; i++){
    b *= 2;
    int bit = (int)b;
    binary_fractional_part[i] = bit + '0';
    b -= bit;
  }
  binary_fractional_part[8] = '\0';
  printf(binary_whole_part);
  printf(".");
  printf(binary_fractional_part);
  printf(".");
}
